Answers.md

```diff
@@ -1,11 +1,12 @@
 # Answers
 
-1.  What is React JS and what problems does it try and solve?
+1.  What is React JS and what problems does it try and solve? React Js is a UI library that allows front ennd developers to develop interactive apps.
 
-1.  What does it mean to _think_ in react?
+1.  What does it mean to _think_ in react?Thinking in react is to create apps using potentially reusable pieces of code called components.
 
 1.  Briefly describe some of the differences between a Class/Stateful component and a Functional/Presentational component.
-
+	Besides syntax, class components has more functionality that are unavailable to functional components like the ability to set state within the component. Although functional components are quicker and easier to write, class components are more powerful.
 1.  Describe state.
-
+state is the part of an app that is suspectible to change and is defined in the constructor of a class components
 1.  Describe props.
+props is data that can be passed between data, however unlike state, props are immutable.

```

Comment:
note that state is immutable also in react, but they give us a method to replace the state object with a new object by giving us the setState() method. we make a new object and swap out the state, never directly mutating it